internal/discoverer/zookeeper_test.go

```diff
@@ -21,29 +18,23 @@ var zkService = "svc"
 var zkNode1 = "{\"host\":\"127.0.0.1:1980\",\"weight\":100}"
 var zkNode2 = "{\"host\":\"127.0.0.1:1981\",\"weight\":100}"
 var zkConn *zk.Conn
+var zkYamlConfig = `
+hosts:
+  - "127.0.0.1:2181"
+prefix: /zookeeper
+weight: 100
+timeout: 10
+`
 
 func GetZkConfig(t *testing.T) *conf.Zookeeper {
-	wd, _ := os.Getwd()
-	dir := wd[:strings.Index(wd, "internal")]
-	content, err := ioutil.ReadFile(dir + "conf/conf.yaml")
-	assert.Nil(t, err)
-	config := &conf.Config{}
-	err = yaml.Unmarshal(content, config)
-	assert.Nil(t, err)
 	zkConf := &conf.Zookeeper{}
-	for n, c := range config.Discovery {
-		if n == zkName {
-			raw, err := yaml.Marshal(c)
-			assert.Nil(t, err)
-			err = yaml.Unmarshal(raw, &zkConf)
-			assert.Nil(t, err)
-			break
-		}
-	}
+	err := yaml.Unmarshal([]byte(zkYamlConfig), zkConf)
+	assert.Nil(t, err)
```

Comment:
Just return err, and assert in where the function is called？

Reply:
It only takes one time to make an assertion here. Is it redundant to judge by the repetition of the return?